Storage/Storage/Tools/GeneralAppSettingsStorage.swift

```diff
@@ -17,8 +18,13 @@ public struct GeneralAppSettingsStorage {
     ///
     private static let refreshSubject = CurrentValueSubject<Void, Never>(())
 
-    public init(fileStorage: FileStorage = PListFileStorage()) {
+    public init(fileStorage: FileStorage = PListFileStorage(), fileURL: URL? = nil) {
         self.fileStorage = fileStorage
+        if let fileURL = fileURL {
+            self.fileURL = fileURL
+        } else {
+            self.fileURL = Constants.generalAppSettingsFileURL
+        }
```

Comment:
Nitpick unrelated to the Modules scaffolding that's the focus on this PR, but why not just write this instead?
```suggestion
        self.fileURL = fileURL ?? Constants.generalAppSettingsFileURL
```